sympy/logic/tests/test_lra_theory.py

```diff
@@ -0,0 +1,505 @@
+
+
+
+        if False in constraints or True in constraints:
+            continue
+
+        phi = And(*constraints)
+        if phi == False:
+            continue
+        cnf = CNF.from_prop(phi); enc = EncodedCNF()
+        enc.from_cnf(cnf)
+        assert all(0 not in clause for clause in enc.data)
+
+        start = time.time()
+        lra, _ = LRASolver.from_encoded_cnf(enc, testing_mode=True)
+        s_subs = lra.s_subs
+        end = time.time()
+        from_encoded_time += end - start
+
+        lra.run_checks = True
+        s_subs_rev = {value: key for key, value in s_subs.items()}
+        lits = {lit for clause in enc.data for lit in clause}
+
+        bounds = [(lra.enc_to_boundary[l], l) for l in lits if l in lra.enc_to_boundary]
+        bounds = sorted(bounds, key=lambda x: (str(x[0].var), x[0].bound, str(x[0].upper))) # to remove nondeterminism
+
+        start = time.time()
+        for b, l in bounds:
+            if lra.result and lra.result[0] == False:
+                break
+            #print("var:", b.var, "bound:", b.bound, "upper:", b.upper, "strict:", b.strict)
+            lra.assert_lit(l)
+            #print(lra.assign, lra.lower, lra.upper)
+
+        end = time.time()
+        assert_time += end - start
+
+        start = time.time()
+        feasible = lra.check()
+        end = time.time()
+        check_time += end - start
+
+        if feasible[0] == True:
+            feasible_count += 1
+            assert check_if_satisfiable_with_z3(constraints) is True
+            cons_funcs = [cons.func for cons in constraints]
+            assignment = feasible[1]
+            assignment = {key.var : value for key, value in assignment.items()}
+            if not (StrictLessThan in cons_funcs or StrictGreaterThan in cons_funcs):
+                assignment = {key: value[0] for key, value in assignment.items()}
+                for cons in constraints:
+                    assert cons.subs(assignment) == True
+
+            else:
+                rat_assignment = find_rational_assignment(constraints, assignment)
+                assert rat_assignment is not None
+        else:
+            assert check_if_satisfiable_with_z3(constraints) is False
+
+            conflict = feasible[1]
+            assert len(conflict) >= 2
+            conflict = {lra.enc_to_boundary[-l].get_inequality() for l in conflict}
+            conflict = {clause.subs(s_subs_rev) for clause in conflict}
+            assert check_if_satisfiable_with_z3(conflict) is False
+
+            # check that conflict clause is probably minimal
+            for subset in itertools.combinations(conflict, len(conflict)-1):
+                assert check_if_satisfiable_with_z3(subset) is True
+    # print("check", check_time)
+    # print("from_encoded", from_encoded_time)
+    # print("assert", assert_time)
+    # print("total", check_time+from_encoded_time+assert_time)
+    # print("sat count", feasible_count)
```

Comment:
```suggestion
```

Reply:
Also remove all the timing stuff above that this is coming from.

If we want to track the performance here, we should add benchmarks to the benchmarks suite https://github.com/sympy/sympy_benchmarks.